### PR TITLE
Added stock span problem

### PR DESCRIPTION
**Pull request by gargVader**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Stock span problem is an interesting application of the stack data structure

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#1566

### CODE REVIEW

File: C-Plus-Plus/other/Stock_span_problem.cpp [hunk 1]
```diff
@@ -0,0 +1,86 @@
+/*
+Stock Span Problem
+===================
+You are given the price of a stock for N consecutive days and are required to find the span
+of stock's price on ith day. The span of a stoock's price on a given day i, is the maximum
```
Comment (atarax665) [inline C-Plus-Plus/other/Stock_span_problem.cpp]:
```suggestion
of stock's price on ith day. The span of a stock's price on a given day i, is the maximum
```

File: C-Plus-Plus/other/Stock_span_problem.cpp [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+consecutive days before the (i+1)th day, for which the stock's price is less than equal to that on
+the ith day.
+
```
Comment (atarax665) [inline C-Plus-Plus/other/Stock_span_problem.cpp:8]: Add link of the problem.

File: C-Plus-Plus/other/Stock_span_problem.cpp [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+		s.push({currSpan, currPrice});
+	}
+
+	// Print ans
+	for (int i = 0; i < n; i++) {
+		cout << span[i] << " ";
+	}
+}
+
+/*
```
Comment (atarax665) [inline C-Plus-Plus/other/Stock_span_problem.cpp:82]: Add space and time complexity of your solution.

**Review by atarax665 — CHANGES_REQUESTED** [review]

**Review by atarax665 — APPROVED** [review]
LGTM :+1:

**Comment by gargVader** [comment]
@shraddhavp Should I add a separate section for stack based problems?

**Comment by shraddhavp** [comment]
yes you can @gargVader

**Comment by shraddhavp** [comment]
create a new issue and do it

**Pull request merged**